modal/_utils/docker_utils.py

```diff
@@ -88,3 +88,24 @@ def valid_dockerignore_file(fp):
     possible_locations.append(context_directory / generic_name)
 
     return next((e for e in possible_locations if valid_dockerignore_file(e)), None)
+
+
+class _AutoDockerIgnore:
+    _custom_repr: Optional[str] = None
+
+    def with_repr(self, custom_repr) -> "_AutoDockerIgnore":
+        # use to give an instance of a matcher a custom name - useful for visualizing default values in signatures
+        self._custom_repr = custom_repr
+        return self
+
+    def __repr__(self) -> str:
+        if self._custom_repr:
+            return self._custom_repr
+
+        return super().__repr__()
+
+    def __call__(self, _: Path) -> bool:
+        return True
+
+
+AUTO_DOCKERIGNORE = _AutoDockerIgnore().with_repr(f"{__name__}.AUTO_DOCKERIGNORE")
```

Comment:
what's all this repr magic used for?

Reply:
Moved the repr inside the class, but it's just a placeholder value for type hints & docs to make it simpler to understand what will happen if you don't pass in a custom ignore pattern-list or callable --> then we default to trying to find a matching dockerignore to your dockerfile (or just a generic one if you're using `docker_commands`) -- same as we have for [add_local_python_source](https://modal.com/docs/reference/modal.Image#add_local_python_source)

<img width="671" alt="Screenshot 2024-12-20 at 13 49 59" src="https://github.com/user-attachments/assets/b8c68f84-98e2-4619-bf98-f715890bea14" />

Reply:
TLDR for why it's needed (this was my suggestion)

It's so that we can use these as default values in signatures and still emit valid python signatures in our type stubs, since things like `inspect.signature(...)` will just use the repr of all default values, and can't figure out stuff like import path of the specific constant we use as a default. 

It's definitely a bit of a hack, but not something users should see. An alternative would be to do a bunch of `ast` magic to extract the signatures.